internal/controllers/user_devices_controller.go

```diff
@@ -164,7 +165,13 @@ func (udc *UserDevicesController) dbDevicesToDisplay(ctx context.Context, device
 		return nil, helpers.GrpcErrorToFiber(err, "failed to get integrations")
 	}
 
+	seen := make([]string, 0)
```

Comment:
What's this `seen` thing? Are we using this?

Reply:
this is how I'm  making sure we don't include duplicates in the final response:

```
	seen := make([]string, 0)
	for _, d := range devices {
		if slices.Contains(seen, d.ID) {
			continue
		}
		seen = append(seen, d.ID)
```